roxie/roxiemem/roxiemem.cpp

```diff
@@ -5212,29 +5215,24 @@ void DataBufferBottom::released()
 
 void DataBufferBottom::noteReleased(const void *ptr)
 {
-    HeapletBase * base = realBase(ptr);
-    if (base == this)
-        DataBufferBase::noteReleased(ptr);
-    else
-        base->noteReleased(ptr);
+    DataBuffer * buffer = queryDataBuffer(ptr);
+    assertex(buffer);
+    buffer->noteReleased(ptr);
 }
 
 void DataBufferBottom::noteLinked(const void *ptr)
 {
-    HeapletBase * base = realBase(ptr);
-    if (base == this)
```

Comment:
It is no longer needed because I always call Link() instead of link(DataBufferBottom *).  It is part of the change to ensure that noteLinked() is only called on real pointers.

The motivation for this is that when the headers and rows are separated the mapping from row->header will not work for header->header, so the two cases need to be treated distinctively.